include/sway/desktop/launcher.h

```diff
@@ -3,10 +3,30 @@
 
 #include <stdlib.h>
 
-struct sway_workspace *workspace_for_pid(pid_t pid);
+struct launcher_ctx {
+	pid_t pid;
+	char *name;
+	struct wlr_xdg_activation_token_v1 *token;
+	struct wl_listener token_destroy;
 
-void launcher_ctx_create(pid_t pid);
+	bool activated;
+
+	struct sway_node *node;
+	struct wl_listener node_destroy;
+
+	struct wl_list link;
+};
+
+struct launcher_ctx *launcher_ctx_find_pid(pid_t pid);
+
+struct sway_workspace *launcher_ctx_get_workspace(struct launcher_ctx *ctx);
 
-void remove_workspace_pid(pid_t pid);
+void launcher_ctx_consume(struct launcher_ctx *ctx);
+
+void launcher_ctx_destroy(struct launcher_ctx *ctx);
+
+const char *launcher_ctx_get_name(struct launcher_ctx *ctx);
```

Comment:
This is unused as well.

Reply:
For reference, this is first used in e8deba2b (launcher: export xdga tokens [...])